dolphin/drivers/dellemc_storage/vmax.py

```diff
@@ -0,0 +1,223 @@
+        """ Connect to storage backend and register using access_info.
+
+        Typical content of the access_info for connecting/registering to VMAX
+
+        access_info = {
+            "storage_id": "",           # Not used in this function
+            "host": "127.0.0.1",        # IP Address of VMAX Storage
+            "port": "8443",             # PORT of VMAX Storage
+            "username": "user",         # Username for VMAX Storage
+            "password": "pass",         # Username for VMAX Storage
+            "extra_attributes" = {
+                "array_id": "00012345", # Array ID
+                "univmax_ver": "90",    # Version of Unishare of VMAX
+                "verify": False         # True/False/Key file path for SSL verify
+            }
+        }
+        """
+
+        # Verify input parameters for registration exists
+        try:
+            self.username = access_info['username']
+            self.password = access_info['password']
+            self.server_ip = access_info['host']
+            self.port = access_info['port']
+
+        except KeyError as err:
+            LOG.error("Invalid input access_info[" + str(err) + "]")
+            return None
+
+        try:
+            # Get optional inputs
+            if 'extra_attributes' in access_info:
+                extra_attr = access_info['extra_attributes']
+
+                if 'verify' in extra_attr:
+                    self.verify = extra_attr['verify']
+                if 'univmax_ver' in extra_attr:
+                    self.univmax_version = extra_attr['univmax_ver']
+                if 'array_id' in extra_attr:
+                    self.symmetrix_id = extra_attr['array_id']
+
+        except KeyError as err:
+            LOG.info("Invalid input access_info[" + str(err) + "]")
+
+        description = 'Unispere array with version '
+        try:
+            # Initialise PyU4V connection to Unisphere
+            conn = PyU4V.U4VConn(
+                u4v_version=self.univmax_version,
+                server_ip=self.server_ip,
+                port=self.port,
+                verify=False,
+                username=self.username,
+                password=self.password)
+
+            # Get the Unisphere version
+            version = conn.common.get_uni_version()
+            description = description + version[0]
+
+            if self.symmetrix_id == "":
+                # Retrieve a list of arrays managed by Unisphere
+                array_list = conn.common.get_array_list()
+
+                if len(array_list) == 0:
+                    LOG.error("Array list from Storage backend is empty")
+                    conn.close_session()
+                    return None
+
+                if len(array_list) > 1:
+                    LOG.info("More than one Storage Array, first one selected")
+
+                self.symmetrix_id = array_list[0]
+
+            # Close the session
+            conn.close_session()
```

Comment:
Close the session in except block too, based upon the check.

Reply:
Moved close to destructor, as we need to keep conn open.